player.go

```diff
@@ -69,26 +69,44 @@ func (a *API) GetDevices() ([]*Device, error) {
 
 // Play starts a new context or resume current playback on the user's active device.
 // https://developer.spotify.com/documentation/web-api/reference/#endpoint-start-a-users-playback
-func (a *API) Play(deviceID string, uris ...string) error {
+// contextURI added to allow for album and playlist support - A blank context uri will play just the track uri.
+func (a *API) Play(deviceID, contextURI string, uris ...string) error {
 	v := make(url.Values)
 	if deviceID != "" {
 		v.Add("device_id", deviceID)
 	}
 
-	if len(uris) == 0 {
+	if (len(uris) == 0) && (len(contextURI) == 0) {
 		return a.put("v1", "/me/player/play", v, nil)
 	}
 
-	body := &struct {
-		URIs []string `json:"uris"`
-	}{URIs: uris}
+	// If I don't put the return into the if statement the IDE states that the body struct will not be used and returned?
```

Comment:
I just remembered, there's a convenient JSON tag we could use here called `omitempty`. We can define our body like:

```
body := &struct {
    ContextURI string `json:"context_uri,omitempty"`
    URIs string `json:"uris,omitempty"`
}{}
```

Then, in the `if` statement, set either `body.ContextURI = contextURI` or `body.URIs = uris`

Reply:
Considering I had done a bunch of `omitempty` yesterday that is embarrassing - I will do it that way.